src/core/config/srp_client.h

```diff
@@ -85,6 +85,30 @@
 #define OPENTHREAD_CONFIG_SRP_CLIENT_SWITCH_SERVER_ON_FAILURE 1
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_SRP_CLIENT_DISALLOW_SERVER_SWITCH_WITH_REGISTERED_HOST
+ *
+ * Define to 1 to disallow SRP client to switch server (when auto-start is used) on failure if host (or any services)
+ * were previously successfully registered with the currently selected server.
+ *
+ * This config is only applicable when `OPENTHREAD_CONFIG_SRP_CLIENT_SWITCH_SERVER_ON_FAILURE` is enabled.
+ *
+ * If this is enabled, once SRP client successfully registers with an auto-start selected SRP server, it stays with
+ * the selected SRP server (even if future SRP updates fail or timeout) while the related server entry remains present
+ * in the Thread Network Data. If the entry is removed from Network Data, then SRP client will try to select a new
+ * server.
+ *
+ * Enabling this behavior would be useful allowing more aggressive switch logic (e.g., switch after fewer timeouts
+ * `OPENTHREAD_CONFIG_SRP_CLIENT_MAX_TIMEOUT_FAILURES_TO_SWITCH_SERVER`) while not allowing switch due to temporary
+ * failures on the currently selected server. This can help avoid situation where the old server may still remember the
+ * registered host/service info and client switching to a new server then failing due to name conflict (when new server
+ * probes for the host/service name original server would not allow new registration).
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_SRP_CLIENT_DISALLOW_SERVER_SWITCH_WITH_REGISTERED_HOST
+#define OPENTHREAD_CONFIG_SRP_CLIENT_DISALLOW_SERVER_SWITCH_WITH_REGISTERED_HOST 1
```

Comment:
```suggestion
#define OPENTHREAD_CONFIG_SRP_CLIENT_DISALLOW_SERVER_SWITCH_WITH_REGISTERED_HOST 0
```
Suggest allowing by default.

Reply:
I think we want to disable this (set it to zero) only if/when we use infra provided SRP servers (where the entry in network data may not disappear).  For default case it may better to keep it enabled.

Reply:
SGTM.